### ISSUE TITLE
Dataset mode initialization issue

### ISSUE THREAD

**Issue by sunnytt98** [zqhang/AnomalyCLIP#89]
Shouldn’t the dataset type be set to **“train”** during training, since the default is **“test”**?
like this
 `train_data = Dataset(root=args.train_data_path, transform=preprocess, target_transform=target_transform, dataset_name = args.dataset, mode='train')
`

https://github.com/zqhang/AnomalyCLIP/blob/3911738c0867544f545a076ad78f3f11d9ecbfdf/train.py#L35C5-L35C138

**Comment by zqhang** [comment]
In test mode, the model uses the test split to fine-tune AnomalyCLIP, since only the test split has ground truth. We train on the test split of one dataset and evaluate on the test split of another dataset. If you switch to train mode, the model will read from the train folder.
